src/base/QXmppStun.cpp

```diff
@@ -2439,8 +2439,8 @@ QList<QUdpSocket*> QXmppIceComponent::reservePorts(const QList<QHostAddress> &ad
 
         // cleanup if we failed
         if (sockets.size() != expectedSize) {
-            for (int i = 0; i < sockets.size(); ++i)
-                delete sockets[i];
+            for (auto &socket : sockets)
+                delete socket;
```

Comment:
`qDeleteAll(sockets);`

Reply:
ditto

Reply:
We actually can't use `std::unique_ptr`s, because then we would need to change the return type and so the public API. `reservePort()` is used in here and it wouldn't make sense to change its return type.